mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -1740,3 +1741,294 @@ Value mlir::rock::addPassThroughIndices(OpBuilder &b, Value transformed,
   }
   return ret;
 }
+
+enum DimType { Upper = 0, Lower = 1 };
+
+struct TransformAttrArgs {
```

Comment:
Please add docs for this.
I d also like a printer added here to support <<

Reply:
Done, https://github.com/ROCm/rocMLIR/blob/851c359837641e3e66490414760e5b64b69a51d4/mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp#L1766-L1788